tracking/src/main/java/org/hps/recon/tracking/kalman/KalmanInterface.java

```diff
@@ -1058,7 +1077,24 @@ private boolean fillAllMeasurements(EventHeader event) {
                     globalX.print("globalX");
                     globalY.print("globalY");
                 }
-                Measurement m = new Measurement(umeas, xStrip, du, time, localHit.getdEdx()*1000000.);
+		List<RawTrackerHit> rawhits =  hit.getRawHits();
+                ShaperFitAlgorithm fitter = new ShaperPileupFitAlgorithm(.5,1);
+                PulseShape shape = new PulseShape.FourPole();
+                fitter.setRunNum(event.getRunNumber());
+                double Variance=0.0;
+                for(RawTrackerHit rth: rawhits){
+                        double Min = 10000;
+                        for(ShapeFitParameters fit : fitter.fitShape(rth, shape)){
+                                if(fit.getT0Err()<Min){
+                                        Min=fit.getT0Err();
+                                }
+                        }
+                        if(module.Layer>1){
+                                Variance+=1/(Min*Min);
+                        }
+                }
+                Variance=1.0/Variance;
+                Measurement m = new Measurement(umeas, xStrip, du, time, localHit.getdEdx()*1000000.,Variance);
```

Comment:
Really, this should be done in clustering but SiTrackerHit doesn't have a place for the error on the time.  We do extend SiTrackerHit in hps-java at org.hps.recon.tracking.SiTrackerHitStrip1D and we could put the error in there but it wouldn't be saved to the LCIO.  For now this is ok...I think we have some other changes to LCIO/lcsim we need to make so we should put this on the list.